content/courses/advanced-ada/chapters/statements.rst

```diff
@@ -3,6 +3,133 @@ Statements
 
 .. include:: ../../global.txt
 
+Simple and Compound Statements
+------------------------------
+
+We can classify statements in simple and compound statements. Simple statements
+don't contain other statements; think of them as "atomic units" that cannot be
+further divided. Compound statements, on the other hand, may contain other
+|mdash| simple or compound |mdash| statements.
+
+Here are some examples from each category:
+
++---------------------+--------------------------------------------------------+
+| Category            | Examples                                               |
++=====================+========================================================+
+| Simple statements   | Null statement, assignment, subprogram call, etc.      |
++---------------------+--------------------------------------------------------+
+| Compound statements | If statement, case statement, loop, block statement    |
++---------------------+--------------------------------------------------------+
+
+.. admonition:: In the Ada Reference Manual
+
+    - `5.1 Simple and Compound Statements - Sequences of Statements <http://www.ada-auth.org/standards/12rm/html/RM-5-1.html>`_
+
+Labels
+~~~~~~
+
+We can use labels to identify statements in the code. They are useful as a way
+to document the code or to indicate important statements in the code.
+
+Labels have the following format: :ada:`<<Some_Label>>`. We write them right
+before the statement we want to apply it to. Let's see an example of labels
+with simple statements:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Statements.Statement_Identifier_1
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Statement_Identifier is
+       pragma Warnings (Off, "is not referenced");
+    begin
+       <<Show_Hello>> Put_Line ("Hello World!");
+       <<Show_Test>>  Put_Line ("This is a test.");
+
+       <<Show_Separator>>
+       <<Show_Block_Separator>>
+       Put_Line ("====================");
+    end Show_Statement_Identifier;
+
+Here, we're labeling each statement. For example, we use the :ada:`Show_Hello`
+label to identify the :ada:`Put_Line ("Hello World!");` statement. Note that we
+can use multiple labels a single statement. In this code example, we use the
+:ada:`Show_Separator` and :ada:`Show_Block_Separator` labels for the same
+statement.
+
+We can use labels with compound statements as well. For example, we can label
+a :ada:`for` loop:
+
+.. code:: ada run_button project=Courses.Advanced_Ada.Statements.Statement_Identifier_2
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    procedure Show_Statement_Identifier is
+       pragma Warnings (Off, "is not referenced");
+
+       Arr   : constant array (1 .. 5) of Integer := (1, 4, 6, 42, 49);
+       Found : Boolean := False;
+    begin
+       <<Find_42>> for E of Arr loop
```

Comment:
Seems better as a loop identifier, IMO.

Reply:
Agreed. My goal is to just show that using labels is also possible in this case. However, I added this phrase:

> Loop statement and block statement identifiers are generally preferred over labels.